src/components/ec-dropdown-search/ec-dropdown-search.spec.js

```diff
@@ -1,40 +1,256 @@
+      text: 'Random Item',
+      disabled: true,
+      disabledReason: 'This reason should not be seen because it\'s overridden by tooltip.content.',
+    };
+
+    const overriddenItem = {
+      ...disabledItem,
+      tooltip: {
+        content: 'Content overridden by item',
+      },
+    };
+
+    const defaultWrapper = mountDropdownSearch({ items: [disabledItem] });
+    const customizedWrapper = mountDropdownSearch({
+      items: [overriddenItem],
+      tooltipOptions: {
+        placement: 'top',
+      },
+    });
+
+    expect(defaultWrapper.element).toMatchDiffSnapshot(customizedWrapper.element);
+  });
+
+  it('should merge given popoverOptions', () => {
+    const defaultWrapper = mountDropdownSearch({});
+    const customizedWrapper = mountDropdownSearch({
+      popoverOptions: {
+        placement: 'top',
+        offset: 16,
+      },
+    });
+
+    expect(defaultWrapper.element).toMatchDiffSnapshot(customizedWrapper.element);
+  });
+
+  describe('filtering', () => {
+    const itemsToFilter = [
+      { text: 'Item ABC' },
+      { text: 'Item BCD' },
+      { text: 'Item cdf' },
+    ];
+
+    function getItemTexts(wrapper) {
+      return wrapper.findAll('.ec-dropdown-search__item').wrappers.map(itemWrapper => itemWrapper.text());
+    }
+
+    it('should filter items', () => {
+      const wrapper = mountDropdownSearch({ items: itemsToFilter });
+      expect(wrapper.findAll('.ec-dropdown-search__item').length).toBe(itemsToFilter.length);
+
+      wrapper.find('.ec-dropdown-search__search-input').setValue('B');
+      expect(getItemTexts(wrapper)).toEqual(['Item ABC', 'Item BCD']);
+    });
+
+    it('should filter items using case insensitive', () => {
+      const wrapper = mountDropdownSearch({ items: itemsToFilter });
+      expect(wrapper.findAll('.ec-dropdown-search__item').length).toBe(itemsToFilter.length);
+
+      wrapper.find('.ec-dropdown-search__search-input').setValue('D');
+      expect(getItemTexts(wrapper)).toEqual(['Item BCD', 'Item cdf']);
+    });
+
+    it('should trim the search text', () => {
+      const wrapper = mountDropdownSearch({ items: itemsToFilter });
+      expect(wrapper.findAll('.ec-dropdown-search__item').length).toBe(itemsToFilter.length);
+
+      wrapper.find('.ec-dropdown-search__search-input').setValue('\t f\n');
+      expect(getItemTexts(wrapper)).toEqual(['Item cdf']);
+    });
+
+    it('should remove the diacritics', () => {
+      const wrapper = mountDropdownSearch({ items: itemsToFilter.map(item => ({ text: item.text.replace('C', 'Č') })) });
+      expect(wrapper.findAll('.ec-dropdown-search__item').length).toBe(itemsToFilter.length);
+
+      wrapper.find('.ec-dropdown-search__search-input').setValue('č');
+      expect(getItemTexts(wrapper)).toEqual(['Item ABČ', 'Item BČD', 'Item cdf']);
+    });
+  });
+
+  describe('events', () => {
+    // .find('stub').trigger('event') does nothing, so we need to emit the event from inside of the stubbed
```

Comment:
can we remove this commented code?

Reply:
It's an explanation why I use `.vm` in the tests, because we should not use it in tests only as a last possible option. I'd like to give more context to the reader why is it here.